frontend/src/employee-frontend/components/unit/tab-unit-information/occupancy/OccupancyDayGraphPlanned.tsx

```diff
@@ -0,0 +1,276 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { Chart, ChartData, ChartOptions, TooltipModel } from 'chart.js'
+import ceil from 'lodash/ceil'
+import isEqual from 'lodash/isEqual'
+import React, { useMemo, useCallback, useState } from 'react'
+import { Line } from 'react-chartjs-2'
+import styled from 'styled-components'
+
+import { formatTime } from 'lib-common/date'
+import { AttendanceReservationReportRow } from 'lib-common/generated/api-types/reports'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import { H3, Label } from 'lib-components/typography'
+import { Gap } from 'lib-components/white-space'
+import colors from 'lib-customizations/common'
+
+import { Translations, useTranslation } from '../../../../state/i18n'
+
+import { ChartTooltip } from './ChartTooltip'
+import { LegendSquare } from './OccupanciesForSingleDay'
+import { DatePoint, defaultChartOptions, hidden, line } from './chart-commons'
+
+interface Props {
+  rows: AttendanceReservationReportRow[]
+}
+
+export default React.memo(function OccupancyDayGraphPlanned({ rows }: Props) {
+  const { i18n } = useTranslation()
+  return (
+    <FixedSpaceColumn>
+      <Graph rows={rows} />
+      <H3>{i18n.unit.occupancy.realtime.legendTitle}</H3>
+      <FixedSpaceColumn>
+        <FixedSpaceRow>
+          <LegendSquare color={colors.grayscale.g100} />
+          <Label>{i18n.unit.occupancy.realtime.children}</Label>
+        </FixedSpaceRow>
+        <FixedSpaceRow>
+          <LegendSquare color={colors.main.m3} />
+          <Label>{i18n.unit.occupancy.realtime.staffRequired}</Label>
+        </FixedSpaceRow>
+      </FixedSpaceColumn>
+    </FixedSpaceColumn>
+  )
+})
+
+const Graph = React.memo(function Graph({ rows }: Props) {
+  const { i18n } = useTranslation()
+  const [tooltipParams, setTooltipParams] = useState<{
+    position?: {
+      x: number
+      y: number
+      xAlign: string
+      yAlign: string
+      caretX: number
+      caretY: number
+    }
+    data?: {
+      date: Date
+      children: number
+      childrenPresent: number
+      staffRequired: number
+    }
+  }>({})
+  const [tooltipVisible, setTooltipVisible] = useState<boolean>(false)
+
+  const showTooltip = useCallback(() => {
+    if (!tooltipVisible) {
+      setTooltipVisible(true)
+    }
```

Comment:
The `if` is unnecessary. `setTooltipVisible(true)` does nothing if it's already true.

Reply:
We also already have a `useBoolean` helper that exposes callbacks like `toggle`, `on` and `off`.

Reply:
Copy-pasted without reading :D I'll fix both

Reply:
https://github.com/espoon-voltti/evaka/pull/5565/commits/cf07f560911655535a236f563179955cb38bce3f